File: Jihun/week_4/9465_스티커.cpp
#include <iostream>
using namespace std;

int main() {
    int n, c;
    cin >> n;
    int arr[2][100000];
    int dp[2][100000];

    for (int i = 0; i < n; i++) {
        cin >> c;
        for (int j = 0; j < 2; j++)
            for (int k = 0; k < c; k++)
                cin >> arr[j][k];
        // 위에 떼는 경우
        dp[0][0] = arr[0][0];
        dp[0][1] = arr[1][0] + arr[0][1];
        // 아래 떼는 경우
        dp[1][0] = arr[1][0];
        dp[1][1] = arr[0][0] + arr[1][1];

        for (int l = 1; l < c; l++) {
            dp[0][l] = arr[0][l] + max(dp[1][l - 1], dp[1][l - 2]);
            dp[1][l] = arr[1][l] + max(dp[0][l - 1], dp[0][l - 2]);
        }

        cout << max(dp[0][c - 1], dp[1][c - 1]) << endl;
    }

    return 0;
}

File: Sihyeon/Week4/9465_스티커.py
t = int(input())

for _ in range(t):
    sticker = []
    n = int(input())
    for _ in range(2):
        arr = list(map(int, input().split()))
        sticker.append(arr)
    dp = [[0] * (n+1) for _ in range(2)]
    dp[0][0] = 0
    dp[1][0] = 0
    dp[0][1] = sticker[0][0]
    dp[1][1] = sticker[1][0]

    for i in range(2, n+1):
        for j in range(2):
            if (j == 0):
                dp[j][i] = max(dp[j+1][i-1], dp[j+1][i-2]) + sticker[j][i-1]
            else:
                dp[j][i] = max(dp[j-1][i-1], dp[j-1][i-2]) + sticker[j][i-1]

    print(max(dp[0][n], dp[1][n]))
